eid_iomem_pci.c

```diff
@@ -0,0 +1,221 @@
+/*
+ * Eidetic IOMEM Device Driver
+ * Copyright (c) 2022, Eideticom
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms and conditions of the GNU General Public License,
+ * version 2, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * Copyright (C) 2022 Eideitcom
+ */
+
+#include <linux/module.h>
+#include <linux/pci.h>
+#include <linux/pci-p2pdma.h>
+#include <linux/fs.h>
+#include <linux/cdev.h>
+#include <linux/pfn_t.h>
+
+#define PCI_VENDOR_EIDETICOM 0x1de5
+#define PCI_DEVICE_IOMEM 0x1000
+
+MODULE_LICENSE("GPL");
+MODULE_AUTHOR("Andrew Maier <andrew.maier@eideticom.com>");
+MODULE_DESCRIPTION("Driver for the Eideticom IOMEM PCIe device");
+
+static int max_devices = 16;
+module_param(max_devices, int, 0444);
+MODULE_PARM_DESC(max_devices, "Maximum number of eid-iomem devices");
+
+static struct class *eid_iomem_class;
+static DEFINE_IDA(eid_iomem_ida);
+static dev_t eid_iomem_devt;
+
+static struct pci_device_id eid_iomem_pci_id_table[] = {
+	{ PCI_DEVICE(PCI_VENDOR_EIDETICOM, PCI_DEVICE_IOMEM), },
+	{ 0, }
+};
+MODULE_DEVICE_TABLE(pci, eid_iomem_pci_id_table);
+
+struct eid_iomem_dev {
+	struct device dev;
+	struct pci_dev *pdev;
+	int id;
+	struct cdev cdev;
+};
+
+static struct eid_iomem_dev *to_eid_iomem(struct device *dev)
+{
+	return container_of(dev, struct eid_iomem_dev, dev);
+}
+
+static int eid_iomem_open(struct inode *inode, struct file *filp)
+{
+	struct eid_iomem_dev *p;
+
+	p = container_of(inode->i_cdev, struct eid_iomem_dev, cdev);
+	filp->private_data = p;
+	//pci_p2pdma_file_open(p->pdev, filp);
```

Comment:
Looks like this has been inadvertently commented out.

Reply:
Oops yup I'm testing on an older kernel still which doesn't have this yet.

Reply:
How could it have pci_p2pmem_mmap() and not pci_p2pdma_file_open()??

Reply:
The [x4 branch](https://github.com/Eideticom/linux-p2pdma/commits/p2pdma_user_cmb_x4) seemed to not have it on this [commit](https://github.com/Eideticom/linux-p2pdma/commit/1e4c3f4b8ed36d8296af64b979bbf68299c65ad0)

Reply:
Oh, stop using that branch completely. It's too old.

Reply:
I am also testing on the [v5 branch](https://github.com/Eideticom/linux-p2pdma) but have been using the x4 to test the hardware until the [issue on v5](https://github.com/Eideticom/linux-p2pdma/issues/1) is fixed.